### PR TITLE
add styles enhancements in various widgets

### PR DESCRIPTION
**Pull request by anserwaseem**
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

**Review by sibteali786 — APPROVED** [review]
thanks for adding styles to dropdown, a much awaited thing @anserwaseem

File: packages/runtime/src/widgets/Card.tsx [hunk 1]
```diff
@@ -41,17 +42,37 @@ export const Card: React.FC<CardProps> = ({ children, styles }) => {
...
+            ${
+              gap
+                ? `
+                    display: block !important;
+                    margin-top: ${gap} !important;
+                  `
+                : ""
+            }
+          }
+        `}
```
Comment (evshi) [inline packages/runtime/src/widgets/Card.tsx:64]: Why is this needed?

File: packages/runtime/src/widgets/Dropdown.tsx [hunk 2]
```diff
@@ -102,25 +129,83 @@ const Dropdown: React.FC<DropdownProps> = (props) => {
...
+          marginBottom: "0px",
+          ...(values?.styles?.visible === false
+            ? { display: "none" }
+            : undefined),
+          ...values?.labelStyle,
+        }}
+      >
+        {values?.id ? (
+          <style>{`
+        [data-testid=${values?.id}] .ant-select .ant-select-selector {
```
Comment (evshi) [inline packages/runtime/src/widgets/Dropdown.tsx:147]: Instead of using the testId, use the stylenames.

File: packages/runtime/src/runtime/body.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export const EnsembleBody: React.FC<EnsembleBodyProps> = ({
     styles: {
       height:
         // We only need to do this calculation if there's a footer otherwise, page can flow freely
-        !isModal && footer
+        isModal && footer
```
Comment (evshi) [inline packages/runtime/src/runtime/body.tsx:54]: This would break the pages where there is a footer, like the Workflow screen

File: packages/runtime/src/widgets/Card.tsx (hunk 1 above)
Comment (evshi) [reply packages/runtime/src/widgets/Card.tsx:64]: In that case, why not just allow setting the styles on the card element directly?

File: packages/runtime/src/widgets/Card.tsx [hunk 4]
```diff
@@ -41,13 +44,17 @@ export const Card: React.FC<CardProps> = ({ children, styles }) => {
...
+
+  const mergedStyles = { ...defaultStyles, ...styles };
   const { shadowOffset, shadowBlur, shadowSpread, shadowColor } = mergedStyles;
+
   return (
     <div
+      className={styles?.names}
       style={{
-        ...mergedStyles,
+        display: "grid",
```
Comment (evshi) [inline packages/runtime/src/widgets/Card.tsx:55]: Why display: grid here?

**Review by evshi — APPROVED** [review]

File: packages/runtime/src/widgets/Card.tsx (hunk 4 above)
Comment (anserwaseem) [reply packages/runtime/src/widgets/Card.tsx:55]: cause default block display isn't letting `gap` property have any effect

Comment (evshi) [reply packages/runtime/src/widgets/Card.tsx:55]: We can use display: flex

**Review by evshi — APPROVED** [review]
